bin/proving-service/src/proxy/status.rs

```diff
@@ -34,31 +30,39 @@ type CachedStatus = ProxyStatusResponse;
 pub struct ProxyStatusService {
     load_balancer: Arc<LoadBalancerState>,
     port: u16,
-    cache: Arc<RwLock<Option<CachedStatus>>>,
+    status_tx: watch::Sender<CachedStatus>,
     update_interval: Duration,
 }
 
 /// Internal gRPC service implementation
 pub struct ProxyStatusGrpcService {
-    cache: Arc<RwLock<Option<CachedStatus>>>,
+    status_rx: watch::Receiver<CachedStatus>,
 }
 
 impl ProxyStatusService {
     pub fn new(load_balancer: Arc<LoadBalancerState>, port: u16) -> Self {
-        let cache = Arc::new(RwLock::new(None));
+        // Create initial empty status
+        let initial_status = ProxyStatusResponse {
+            version: env!("CARGO_PKG_VERSION").to_string(),
+            supported_proof_type: 0, // Will be updated immediately
```

Comment:
Why can't this be read from the load_balancer here yet?

Reply:
I sent a commit creating the initial status from the LB.